sdks/go/pkg/beam/pardo_test.go

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package beam
+
+import (
+	"testing"
+)
+
+func TestRecommendParDo(t *testing.T) {
+	var tests = []struct {
+		name      string
+		outputDim int
+		want      string
+	}{
+		{"zero outputs", 0, "ParDo0"},
+		{"one output", 1, "ParDo"},
+		{"more than 7 outputs", 10, "ParDoN"},
+	}
+
+	for _, tt := range tests {
+		testName := tt.name
+		t.Run(testName, func(t *testing.T) {
+			got := recommendParDo(tt.outputDim)
+			if got != tt.want {
+				t.Errorf("RecommendParDo(%v) = %v, want %v", tt.outputDim, got, tt.want)
+			}
+		})
+	}
+}
+
+func testFunction() int64 {
+	return 42
+}
+
+func TestFormatParDoError(t *testing.T) {
+	got := formatParDoError(testFunction, 2, 1)
```

Comment:
Since the "complicated" bit for this is in the recommendParDo code, which is already being tested very well, this test should be sufficient for the parts in formatParDoError. I don't expect the formatting to change any time soon.

We do have the english grammar nit ("1 outputs"), but handling that grammar correction isn't worth the extra code to resolve it.

Reply:
I'm open to spend a bit more time to do this, or would you like me to remove the test here? 

I agree, covering the the `recommendParDo` function is more important. Happy to defer to your decision here.

Reply:
Please do keep the test! My comment was that because the logic that's unique to formatParDoError is so small, the test can also be small.